File: server/router/admin/dorm_room.go
package admin

import (
	"github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DormRoomRouter struct {
}

// InitDormRoomRouter 初始化 宿舍 路由信息
func (s *DormRoomRouter) InitDormRoomRouter(Router *gin.RouterGroup) {
	dormRoomRouter := Router.Group("dormRoom").Use(middleware.OperationRecord())
	dormRoomRouterWithoutRecord := Router.Group("dormRoom")
	var dormRoomApi = v1.ApiGroupApp.AdminApiGroup.DormRoomApi
	{
		dormRoomRouter.POST("createDormRoom", dormRoomApi.CreateDormRoom)             // 新建宿舍
		dormRoomRouter.DELETE("deleteDormRoom", dormRoomApi.DeleteDormRoom)           // 删除宿舍
		dormRoomRouter.DELETE("deleteDormRoomByIds", dormRoomApi.DeleteDormRoomByIds) // 批量删除宿舍
		dormRoomRouter.PUT("updateDormRoom", dormRoomApi.UpdateDormRoom)              // 更新宿舍
		dormRoomRouter.GET("alloc", dormRoomApi.DormAlloc)
		dormRoomRouter.GET("student", dormRoomApi.GetDormStudents)
		dormRoomRouter.GET("remind", dormRoomApi.Remind)
	}
	{
		dormRoomRouterWithoutRecord.GET("findDormRoom", dormRoomApi.FindDormRoom)       // 根据ID获取宿舍
		dormRoomRouterWithoutRecord.GET("getDormRoomList", dormRoomApi.GetDormRoomList) // 获取宿舍列表
	}
}

File: web/src/api/dormRoom.js
import service from '@/utils/request'

// @Tags DormRoom
// @Summary 创建宿舍
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DormRoom true "创建宿舍"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /dormRoom/createDormRoom [post]
export const createDormRoom = (data) => {
  return service({
    url: '/dormRoom/createDormRoom',
    method: 'post',
    data
  })
}

// @Tags DormRoom
// @Summary 删除宿舍
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DormRoom true "删除宿舍"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /dormRoom/deleteDormRoom [delete]
export const deleteDormRoom = (params) => {
  return service({
    url: '/dormRoom/deleteDormRoom',
    method: 'delete',
    params
  })
}

// @Tags DormRoom
// @Summary 批量删除宿舍
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除宿舍"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /dormRoom/deleteDormRoom [delete]
export const deleteDormRoomByIds = (params) => {
  return service({
    url: '/dormRoom/deleteDormRoomByIds',
    method: 'delete',
    params
  })
}

// @Tags DormRoom
// @Summary 更新宿舍
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DormRoom true "更新宿舍"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /dormRoom/updateDormRoom [put]
export const updateDormRoom = (data) => {
  return service({
    url: '/dormRoom/updateDormRoom',
    method: 'put',
    data
  })
}

// @Tags DormRoom
// @Summary 用id查询宿舍
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query model.DormRoom true "用id查询宿舍"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /dormRoom/findDormRoom [get]
export const findDormRoom = (params) => {
  return service({
    url: '/dormRoom/findDormRoom',
    method: 'get',
    params
  })
}

// @Tags DormRoom
// @Summary 分页获取宿舍列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取宿舍列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /dormRoom/getDormRoomList [get]
export const getDormRoomList = (params) => {
  return service({
    url: '/dormRoom/getDormRoomList',
    method: 'get',
    params
  })
}

export const dormAlloc = (params) => {
  return service({
    url: '/dormRoom/alloc',
    method: 'get',
    params
  })
}

export const getDormStudents = (params) => {
  return service({
    url: '/dormRoom/student',
    method: 'get',
    params
  })
}

export const remind = (params) => {
  return service({
    url: '/dormRoom/remind',
    method: 'get',
    params
  })
}
